src/nvim/option.c

```diff
@@ -4195,6 +4255,21 @@ set_num_option (
 
   options[opt_idx].flags |= P_WAS_SET;
 
+  if (!starting && errmsg == NULL) {
+    char_u buf_old[11], buf_new[11], buf_type[7];
+    vim_snprintf((char *)buf_old, 10, "%ld", old_value);
```

Comment:
A buffer size of `NUMBUFLEN` would be okay. We could put this code in a function and use it for all four use cases in this file.